ocean_lib/models/test/test_datatoken.py

```diff
@@ -97,14 +97,15 @@ def test_main(
     assert not permissions[DatatokenRoles.PAYMENT_MANAGER]
 
     # not erc20 deployer role
+    args = (
+        DatatokenArguments(
+            name="DT1",
+            symbol="DT1Symbol",
+        ),
+        {"from": another_consumer_wallet, "required_confs": 0},
+    )
     with pytest.raises(Exception, match="new data token has no address"):
-        data_nft.create_datatoken(
-            DatatokenArguments(
-                name="DT1",
-                symbol="DT1Symbol",
-            ),
-            {"from": another_consumer_wallet, "required_confs": 0},
-        )
+        confirm_failed(data_nft, "create_datatoken", args, "")
```

Comment:
I also discovered the problem where I didn't remove the contract call in args, below. Please keep all the other changes except this one and let's discuss this particular fix later.

Reply:
why that particular fix could not be fixed here, at least temporary because it will fail further? and why is not a good solution?